web/client/api/GeoStoreDAO.js

```diff
@@ -164,7 +164,7 @@ const Api = {
     },
     updateResourcePermissions: function(resourceId, securityRules) {
         let payload = "<SecurityRuleList>";
-        for (let rule of securityRules.SecurityRuleList.SecurityRule) {
+        (securityRules && securityRules.SecurityRuleList && _.castArray(securityRules.SecurityRuleList.SecurityRule) || []).forEach( rule => {
```

Comment:
let's use a reduce

Reply:
Yes, I didn't want to change this part not covered by a test. going to externalize SecurityRule and test it